src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.Variables.cs

```diff
@@ -0,0 +1,434 @@
+
+            internal void GetMembers(ArrayBuilder<(VariableIdentifier, int)> builder, int containingSlot)
+            {
+                (int id, int index) = DeconstructSlot(containingSlot);
+                var variables = GetVariablesForId(id)!;
+                var variableBySlot = variables._variableBySlot;
+                for (index++; index < variableBySlot.Count; index++)
+                {
+                    var variable = variableBySlot[index];
+                    if (variable.ContainingSlot == containingSlot)
+                    {
+                        builder.Add((variable, ConstructSlot(id, index)));
+                    }
+                }
+            }
+
+            private Variables GetVariablesForVariable(VariableIdentifier identifier)
+            {
+                int containingSlot = identifier.ContainingSlot;
+                if (containingSlot > 0)
+                {
+                    return GetVariablesForId(DeconstructSlot(containingSlot).Id)!;
+                }
+                return GetVariablesContainingVariable(identifier.Symbol);
+            }
+
+            internal Variables GetVariablesContainingVariable(Symbol symbol)
+            {
+                switch (symbol)
+                {
+                    case LocalSymbol:
+                    case ParameterSymbol:
+                    case MethodSymbol { MethodKind: MethodKind.LocalFunction }:
+                        if (symbol.ContainingSymbol is MethodSymbol method &&
+                            GetVariablesForSymbol(method.PartialImplementationPart ?? method) is { } variables)
+                        {
+                            return variables;
+                        }
+                        break;
+                }
+                // Fallback to the outermost scope for the remaining cases. Those cases include: static fields;
+                // variables declared in field initializers; locals and parameters when the root symbol is null;
+                // and error cases such as an instance field referenced in a static method (no containing slot).
+                return GetVariablesRoot();
+            }
+
+            private Variables GetVariablesRoot()
+            {
+                var variables = this;
+                while (variables.Container is { } container)
+                {
+                    variables = container;
+                }
+                return variables;
+            }
+
+            private Variables? GetVariablesForId(int id)
+            {
+                var variables = this;
+                do
+                {
+                    if (variables.Id == id)
+                    {
+                        return variables;
+                    }
+                    variables = variables.Container;
+                }
+                while (variables is { });
+                return null;
+            }
+
+            private Variables? GetVariablesForSymbol(MethodSymbol symbol)
```

Comment:
>GetVariablesForSymbol [](start = 31, length = 21)

I think the name is somewhat confusing, especially that ```Variables GetVariablesContainingVariable(Symbol symbol)``` takes symbol as well. Given the terminology used in doc comments and my understanding of the purpose of this method, I suggest using "GetVariablesForMethodScope"  as the name.